### ISSUE TITLE
doesn't pick the tune

### ISSUE THREAD

**Issue by alekksander** [gstraube/cythara#5]
im having problem to make it work right — it doesn't pick any other tune than G3 (for some reason). even string G is hardly picked up by this app. looks like it's half deaf.

**Comment by gstraube** [comment]
Could you please tell me which device and Android version you are using and what instrument you are trying to tune? If you don't mind, could you also try playing sounds via https://www.fender.com/online-guitar-tuner/electric-guitar-tuning/ (using standard tuning) and/or http://onlinetonegenerator.com/ (with standard guitar tuning and a frequency of e.g. 146 or 110 Hz)?

**Comment by alekksander** [comment]
Thank You for Your interest.
I'm using Cat S40 with Android 5.1. I've used (non foss) tuners before and never had this problem (if anything other tuners were too sensitive and pointed tune from ambient — in such case i suggested gain control, but never got positive response, hence my interest in Your project :) ).

i played tunes from fender's site — the result is better than with classical guitar i tried but still so so… app picks up notes only at first, after a moment it looses it or sometimes picks up note, looses it and then for the moment catches it again. also i've not once seen this app to show no additional cents (never points exact 0). 

ps. would it be possible to make it work smoother? perhaps some kind of animation between

**Comment by gstraube** [comment]
Don't be sorry, I appreciate your feedback! I'll try to increase the sensitivity (I agree that it is not really satisfactory at the moment) and improve accuracy. Adjusting the sampling rate should reduce the lag. A smoother transition would indeed look better.

**Comment by gstraube** [comment]
There is a new release (https://github.com/gstraube/cythara/releases/tag/v1.6). Could you please try it out when you have time? Thank you in advance!

**Comment by alekksander** [comment]
sry for late reply, i don't have a lot of spare time lately. it works somewhat better, but i still can't tune my guitar with it :( on fender's site for instance, one of the notes (i think it was d) appears for a sec as out of tune, and then nothing happens…

**Comment by gstraube** [comment]
No problem! It works in a simulator running Android 5.1 so it does not seem to be a software issue. Maybe some parameters of the audio recorder need to be different in order for it to work properly with the Cat S40. I'll look into it.

**Comment by alekksander** [comment]
oh, that's an interesting way to test it. do You pass internal signal to the app, or go through speakers and microphone?

**Comment by gstraube** [comment]
I ran the emulator on my notebook and used its microphone. Regarding real hardware, I only have two devices at my disposal (my phone and my notebook).

The attached version should detect the pitch quicker (I removed the custom pitch sampling completely). Please let me know if this improves things for you.

[cythara.zip](https://github.com/gstraube/cythara/files/1427578/cythara.zip)

**Comment by alekksander** [comment]
will test tomorrow or later this week.
meanwhile: :beer:

**Comment by alekksander** [comment]
it works better for me, but it's still not a 100% reliable tool for tuning.

**Comment by gstraube** [comment]
I'm sorry to hear that! Could tell me the names of some tuners which work on your phone? Maybe their implementation could provide some clues (although those weren't FOSS apps, right?).

**Comment by alekksander** [comment]
i'll get into that with pleasure, but now i don't have time to make a proper research, sorry. (i've tried some from f-droid (from top of my head — practice hub, guitar tuner (maybe it was on xda, but foss), also accordion tuner `https://fossdroid.com/a/tuner.html` (don't remember if it worked)))

**Comment by gstraube** [comment]
Sure, no problem, it's not that urgent. I'll have a look at the apps you mentioned. Thanks!

**Comment by gstraube** [comment]
Is this the Guitar Tuner you mentioned: https://github.com/chRyNaN/Android-Guitar-Tuner?

**Comment by alekksander** [comment]
that one i've never seen before :)

here it is:
[https://github.com/andryr/guitar-tuner](url)

i also remember trying:
[https://play.google.com/store/apps/details?id=com.darshancomputing.tuner](url)

but to be honest i can't recall which one was fine, which wasn't. i can download them all and try out again with some short depiction. but lately i experienced what does it mean to lack of enough free time, so not today, sorry.

**Comment by gstraube** [comment]
You don't have to do that. That's already very helpful. Thank you!

**Comment by gstraube** [comment]
I tried the two apps you mentioned and the one I found. On my phone the Noteworthy Tuner (https://play.google.com/store/apps/details?id=com.darshancomputing.tuner) worked best. Interestingly, it uses the same library for pitch detection (but with a lower sampling rate). I tinkered with the parameters (sampling rate, buffer size, buffer overlap) but it did not make a great difference concerning sensitivity.

Maybe the Noteworthy Tuner works better for you? In case you haven't seen it already, there is also a Github page with APK releases: https://github.com/darshan-/Noteworthy-Tuner. 

Sorry for the late and unhelpful reply!
